app/pods/agendas/controller.js

```diff
@@ -7,9 +7,54 @@ import { action } from '@ember/object';
 import DefaultQueryParamsMixin from 'ember-data-table/mixins/default-query-params';
 
 export default class AgendasController extends Controller.extend(DefaultQueryParamsMixin) {
+  @service router;
   @service currentSession;
 
   @tracked isCreatingNewSession = false;
+  @tracked dateFilter = '';
+  @tracked sort = '-status,created-for.planned-start';
+  @tracked from = null;
+  @tracked to = null;
+  @tracked page = 0;
+
+  queryParams = ['from', 'to'];
+  dateRegex = /^(?:(\d{1,2})-)??(?:(\d{1,2})-)?(\d{4})$/;
+
+  @action
+  setDateFilter(date) {
+    const newDate = date.split('/').join('-');
+    const match = this.dateRegex.exec(newDate);
+
+    if (!match) {
+      this.from = null;
+      this.to = null;
+      return;
```

Comment:
Persoonlijk vind ik het hier duidelijker/logischer om met if-else te werken ipv met early-return.
 
Als we `this.from` en `this.to` aanpassen (en dus model refresh triggeren), kunnen we `this.page` best ook naar 0 resetten.